activation/malfeasance2.go

```diff
@@ -3,16 +3,96 @@ package activation
 import (
 	"context"
 	"fmt"
+	"sync"
+
+	"go.uber.org/zap"
 
 	"github.com/spacemeshos/go-spacemesh/activation/wire"
 	"github.com/spacemeshos/go-spacemesh/codec"
 	"github.com/spacemeshos/go-spacemesh/common/types"
+	"github.com/spacemeshos/go-spacemesh/log"
+	"github.com/spacemeshos/go-spacemesh/signing"
 )
 
-type MalfeasanceHandlerV2 struct{}
+type MalfeasanceHandlerV2 struct {
+	logger *zap.Logger
+
+	malPublisher malfeasancePublisher
+	edVerifier   *signing.EdVerifier
+	validator    nipostValidatorV2
+
+	smeshingMutex sync.Mutex
```

Comment:
NIT (it guards the `singers` field):
```suggestion
	signersMutex sync.Mutex
```